apps/test/solutions/applab/ec_setprop.js

```diff
@@ -0,0 +1,224 @@
+      xml:
+        'setProperty("my_image", "width", 11);' +
+        'setProperty("my_image", "height", 12);' +
+        'setProperty("my_image", "x", 13);' +
+        'setProperty("my_image", "y", 14);' +
+        'setProperty("my_image", "picture", "' + facebookImage + '");' +
+        'setProperty("my_image", "hidden", true);',
+      runBeforeClick: function (assert) {
+        // add a completion on timeout since this is a freeplay level
+        testUtils.runOnAppTick(Applab, 2, function () {
+          var image = document.getElementById('my_image');
+
+          // we set style.width/height instead of the width/height attributes
+          assert.equal(image.style.width, '11px');
+          assert.equal(image.style.height, '12px');
+          assert.equal(image.getAttribute('width'), null);
+          assert.equal(image.getAttribute('height'), null);
+
+          assert.equal(image.style.left, '13px');
+          assert.equal(image.style.top, '14px');
+
+          assert(/facebook_purple.png$/.test(image.src));
+
+          // visibility is set via a class, so use getComputedStyle
+          assert(window.getComputedStyle(image).visibility, 'hidden');
+
+          Applab.onPuzzleComplete();
+        });
+      },
+      customValidator: function (assert) {
+        // No errors in output console
+        var debugOutput = document.getElementById('debug-output');
+        assert.equal(debugOutput.textContent, "");
+        return true;
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      }
+    },
+
+    {
+      description: "setProperty on API created dropdown",
+      editCode: true,
+      xml:
+        'dropdown("my_drop", "option1", "option2");' +
+        'setProperty("my_drop", "options", ["one", "two", "three"]);',
+      runBeforeClick: function (assert) {
+        // add a completion on timeout since this is a freeplay level
+        testUtils.runOnAppTick(Applab, 2, function () {
+          var dropdown = $("#my_drop");
+
+          assert.equal(dropdown.children().length, 3);
+          assert.equal(dropdown.children().eq(0).text(), 'one');
+          assert.equal(dropdown.children().eq(1).text(), 'two');
+          assert.equal(dropdown.children().eq(2).text(), 'three');
+
+          Applab.onPuzzleComplete();
+        });
+      },
+      customValidator: function (assert) {
+        // No errors in output console
+        var debugOutput = document.getElementById('debug-output');
+        assert.equal(debugOutput.textContent, "");
+        return true;
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      }
+    },
+
+    {
+      description: "setProperty on design mode created dropdown",
```

Comment:
I spy a bug in the "remove any extra options" logic for dropdowns -- if the new list is 2 or more elements shorter than the old one, not all of the old elements will be removed. Maybe change one of these tests to cover that case?